src/main/java/io/numaproj/numaflow/sideinput/SideInputRetriever.java

```diff
@@ -0,0 +1,15 @@
+package io.numaproj.numaflow.sideinput;
+
+
+/**
+ * SideInputRetriever exposes method for retrieving side input.
+ * Implementations should override the retrieveSideInput method
+ * which will be used for updating the side input.
+ */
+public abstract class SideInputRetriever {
+    /**
+     * method which will be used for retrieving side input.
+     * @return Message which contains side input
```

Comment:
what's the behaviour when exceptions are encountered during the `retrieveSideInput()` call? If the contract is to always return a noBroadcastMessage, then please add the contract to the abstract method description so that the implementors will follow. Let's always expose ALL the important pieces of information of a contract to implementors.